### ISSUE TITLE
If a conversation answer contains a keyword that a hears() function is listening for, it will retrigger the conversation

### ISSUE THREAD

**Issue by davetosh** [mpociot/botman#52]
I have a simple conversation that moves through a few questions for the user. Everything works well unless one of the answers contains the word 'this', if that happens, then the conversation goes back to the beginning.

Steps to replicate:

Take the following example

```
class PizzaConversation extends Conversation
{
    protected $size;

    public function askSize()
    {
        $this->ask('What pizza size do you want?', function(Answer $answer) {
            // Save size for next question
            $this->size = $answer->getText();

            $this->say('Got it. Your pizza will be '.$answer->getText());

            $this->askForToppings();
        });
    }

    public function askForToppings()
    {
        $this->ask('What toppings would you like?', function(Answer $answer) {

            $this->say('Got it. Your pizza will have ' . $answer->getText() . ' on it');

        });
    }

    public function run()
    {
        // This will be called immediately
        $this->askSize();
    }
}
```

If the answers are 'large' and then 'cheese and pineapple' then everything works as expected.

However, if the first answer was something like 'make this large' then the conversation will trigger the second question and loop back to the beginning asking the first question again. 

If the conversation has 4 or 5 questions, at any point if the answer contains the word 'this' it always cycles back to the first question again.

**Comment by davetosh** [comment]
Testing a little more, it seems that any word in the answer containing _'thi'_ will cycle the conversation back to the beginning. In the example above - if someone was to answer _'I'm still thinking'_, the second question will trigger (which is as expected) but the conversation then goes back to the beginning and asks the first question again before the user has answered the second question.

**Comment by davetosh** [comment]
Sorry for all the messages. I think I now understand the issue but am unsure if it is a bug or intended behaviour.

If a conversation answer contains a keyword that a hears() function is listening for, it retriggers the conversation. So, in my first example the reason words containing _'thi'_ kept restarting the conversation is because one of the hears functions was listening for _'hi'_. 

`$botman->hears('hi', function (BotMan $bot) {
        $bot->startConversation(new OrderPizza());
});`

**Comment by mpociot** [comment]
Hello,

Thanks for reporting this bug.
There were two issues:
- The regular expression, that checked the incoming messages, was matching everywhere in the string.
- Conversation answers could re-trigger `hears` commands

It's fixed in release `1.0.1`

**Comment by davetosh** [comment]
Thanks for fixing this so quickly.
